appserver/ejb/ejb-rest-remoting/client/src/main/java/fish/payara/ejb/rest/client/RemoteEJBContextFactory.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) 2019 Payara Foundation and/or its affiliates. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of either the GNU
+ * General Public License Version 2 only ("GPL") or the Common Development
+ * and Distribution License("CDDL") (collectively, the "License").  You
+ * may not use this file except in compliance with the License.  You can
+ * obtain a copy of the License at
+ * https://github.com/payara/Payara/blob/master/LICENSE.txt
+ * See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing the software, include this License Header Notice in each
+ * file and include the License file at glassfish/legal/LICENSE.txt.
+ *
+ * GPL Classpath Exception:
+ * The Payara Foundation designates this particular file as subject to the "Classpath"
+ * exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.ejb.rest.client;
+
+import java.util.Hashtable;
+
+import javax.naming.Context;
+import javax.naming.NamingException;
+import javax.naming.spi.InitialContextFactory;
+
+/**
+ * This is the context factory that creates the context used for looking up and invoking
+ * remote EJBs.
+ * 
+ * <p>
+ * Clients wanting to use remote EJB in this way should set the property in the initial context
+ * environment as follows:
+ * 
+ * <pre>
+ * <code>
+ *   Hashtable&lt;String, String&gt; environment = new Hashtable&lt;&gt;();
+ *   environment.put(INITIAL_CONTEXT_FACTORY, "fish.payara.ejb.rest.client.RemoteEJBContextFactory");
+ *    ...
+ *   new InitialContext(environment);
+ * <code>
+ * </pre>
+ * 
+ * @author Arjan Tijms
+ * @since Payara 5.191
+ */
+public class RemoteEJBContextFactory implements InitialContextFactory {
+    
+    public static final String FISH_PAYARA_WITH_CONFIG = "fish.payara.withConfig";
```

Comment:
Are these relevant outside this package?

Reply:
>Are these relevant outside this package?

Yes, they can, and perhaps should, be used to set properties in the environment map.

See e.g.

```
Properties props = new Properties(); 
props.put(Context.INITIAL_CONTEXT_FACTORY, JBOSS_CONTEXT); 
props.put(Context.PROVIDER_URL, "remote://localhost:4447"); 
props.put(Context.SECURITY_PRINCIPAL, "testuser"); 
props.put(Context.SECURITY_CREDENTIALS, "testpassword"); 
InitialContext initialContext = new InitialContext(props); 
```

https://coderanch.com/t/593853/java/JNDI-lookup-GlassFish-JBoss